firm/Firm.java

```diff
@@ -0,0 +1,34 @@
+package firm;
+
+public abstract class Firm{
+    private final String name;
```

Comment:
Can firm have salary or position? Possibly, Worker or something like this should be more suitable

Reply:
Or, if you want to work with their classes as positions, we can name it Position and rename interface to PositionPinter or something like this